src/secp256k1/src/cpp/Scalar.cpp

```diff
@@ -281,19 +281,14 @@ std::string Scalar::GetHex() const {
     return ss.str();
 }
 
-void Scalar::SetHex(const std::string& str) const {
+void Scalar::SetHex(const std::string& str) {
     unsigned char buffer[32];
 
-    for (int i = 0; i < 32; i+=2)
-        buffer[i] =  str[i] * 16 + str[i + 1];
-
-    int overflow = 0;
-
-    secp256k1_scalar_set_b32(reinterpret_cast<secp256k1_scalar *>(value_), buffer, &overflow);
-
-    if (overflow) {
-        throw "Scalar: decoding overflowed";
+    for (int i = 0; i < 32; i++) {
```

Comment:
Same as previous issue that we should not hard-coded buffer size.

Reply:
make sense